src/ui/common/src/components/integrations/connectedIntegrations.tsx

```diff
@@ -63,6 +70,23 @@ export const ConnectedIntegrations: React.FC<ConnectedIntegrationsProps> = ({
         connectedIntegrationType
     )
   );
+
+  // For each integration, count the number of workflows that use it.
+  // Fetch the number of workflows for each integration.
+  const {
+    data: workflowsByIntegration,
+    error: fetchWorkflowsError,
+    isLoading,
+  } = useIntegrationsWorkflowsGetQuery({ apiKey: user.apiKey });
+  if (isLoading) {
+    return <CircularProgress />;
+  }
+  if (fetchWorkflowsError) {
+    setShowWorkflowsFetchErrorToast(true);
```

Comment:
@agiron123 should we do this in a `useEffect` here? I feel we should only manipulate states in hooks or in component handlers like `onChange`, but I couldn't explain the reason well :)

Reply:
Yes, this should be done inside of a useEffect so that we can have proper react lifecycle. 

```
useEffect(() => {
  if (fetchWorkflowsError) {
    setShowWorkflowsFetchErrorToast(true);
  } else {
    // remember to hide if this goes away.
    setShowWorkflowsFetchErrorToast(false);
  }
}, [fetchWorkflowsError]);
```